mne/viz/backends/_vispy.py

```diff
@@ -0,0 +1,560 @@
+        faces = np.array(surface['tris']).astype(np.uint32)
+        if colormap is not None and scalars is not None:
+            cm = Colormap(colormap / 255.0)
+            if vmin is None:
+                vmin = min(scalars)
+            if vmax is None:
+                vmax = max(scalars)
+            nscalars = (scalars - vmin) / (vmax - vmin)
+            vcolors = cm.map(nscalars)
+            mesh = scene.visuals.Mesh(vertices=vertices,
+                                      faces=faces,
+                                      vertex_colors=vcolors,
+                                      shading='smooth',
+                                      parent=self.view.scene)
+        else:
+            mesh = scene.visuals.Mesh(vertices=vertices,
+                                      faces=faces,
+                                      color=color, parent=self.view.scene,
+                                      shading='smooth')
+        if mesh is not None:
+            mesh.shininess = default_mesh_shininess
+            mesh.attach(Alpha(opacity))
+            if backface_culling:
+                mesh.set_gl_state(cull_face=True)
+        if offset is not None:
+            mesh.set_gl_state(depth_test=True,
+                              polygon_offset=(offset, offset),
+                              polygon_offset_fill=True)
+
+    def sphere(self, center, color, scale, opacity=1.0,
+               resolution=8, backface_culling=False):
+        """Add sphere in the scene.
+
+        Parameters
+        ----------
+        center: ndarray, shape(n_center, 3)
+            The list of centers to use for the sphere(s).
+        color: tuple
+            The color of the sphere(s): (red, green, blue).
+        scale: float
+            The scale of the sphere(s).
+        opacity: float
+            The opacity of the sphere(s).
+        backface_culling: bool
+            If True, enable backface culling on the sphere(s).
+        """
+        from vispy.geometry import create_sphere
+        from vispy.util.transforms import translate
+
+        # fetch original data
+        meshdata = create_sphere(radius=scale * default_sphere_radius,
+                                 cols=resolution, rows=resolution)
+        orig_vertices = meshdata.get_vertices()
+        orig_faces = meshdata.get_faces()
+        nvertices = len(orig_vertices)
+        nfaces = len(orig_faces)
+        ncenter = len(center)
+        voffset = 0
+        foffset = 0
+
+        # accumulate mesh data
+        acc_vertices = np.empty((ncenter * nvertices, 3), dtype=np.float32)
+        acc_faces = np.empty((ncenter * nfaces, 3), dtype=np.uint32)
+        for c in center:
+            # apply index shifting and accumulate faces
+            current_faces = orig_faces + voffset
+            acc_faces[foffset:foffset + nfaces, :] = current_faces
+            foffset += nfaces
+
+            # apply translation and accumulate vertices
+            mat = translate(c).T
```

Comment:
can you translate all centers with one call to translate?

Reply:
This creates a 4x4 translation matrix depending on the given vector, it's not actually translating anything. I use later on in the code to apply the translation on the vertices of the sphere. Each sphere has a different center, so each matrix will be different. 

I thik I could leverage the fact that it's only a translation to apply the translation after the `for` loop for example but I'm not enterely sure if we win a lot that way, I'll need to benchmark this.